### PR TITLE
Refactor job status resource

### PR DESCRIPTION
**Pull request by jonathanstowell**

### CODE REVIEW

File: test/ZendeskApi.Client.Tests/Resources/JobStatusResourceTests.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        }
+
+        [Fact]
+        public async Task ListAsync_WhenCalled_ShouldGetAll()
+        {
+            var results = await _resource.ListAsync();
+
+            Assert.Equal(100, results.Count);
+
+            for (var i = 1; i <= 100; i++)
```
Comment (LoadData) [inline test/ZendeskApi.Client.Tests/Resources/JobStatusResourceTests.cs]: We have this same loop in almost all the new test classes. Can we pull it out into it's own method we can reuse?

File: test/ZendeskApi.Client.Tests/Resources/JobStatusResourceTests.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        }
+
+        [Fact]
+        public async Task GetAsync_WhenCalledWithIds_ShouldGetAll()
+        {
+            var results = await _resource.GetAsync(new string[] { "1", "2", "3" });
+
+            Assert.Equal(3, results.Count);
+
+            for (var i = 1; i <= 3; i++)
```
Comment (LoadData) [inline test/ZendeskApi.Client.Tests/Resources/JobStatusResourceTests.cs]: If made generic enough, we'd be able to use that here too.

File: test/ZendeskApi.Client.Tests/Resources/JobStatusResourceTests.cs (hunk 1 above)
Comment (jonathanstowell) [reply test/ZendeskApi.Client.Tests/Resources/JobStatusResourceTests.cs]: Any ideas on how? The model we assert against is different every time. You could create a func to pass in the assertions but it kinda feels a bit pointless just hiding a for loop?

Comment (LoadData) [reply test/ZendeskApi.Client.Tests/Resources/JobStatusResourceTests.cs]: Oh yeah, that's true. Probably not much point then.

**Pull request merged**